src/ng/filter/filters.js

```diff
@@ -331,11 +331,15 @@ function formatNumber(number, pattern, groupSep, decimalSep, fractionSize) {
   }
 }
 
-function padNumber(num, digits, trim) {
+function padNumber(num, digits, trim, negWrap) {
   var neg = '';
   if (num < 0) {
-    neg =  '-';
-    num = -num;
+    if (negWrap) {
+      num = -num + 1;
```

Comment:
Sorry, I misunderstood your original point.

> I just realized: what if G is not included in the format string ?
> Then -0026-01-01 will be January 1st, 27, right ?

Yes, but I would say this is expected. If there is a need to reflect BC years as 0/negative, then we have to add support for the pattern *u...*